events.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"time"
+)
+
+type PeerDialLog struct {
+	Dials    []DialAttempt `json:"dials"`
+	Success  bool          `json:"success"`
+	Duration string        `json:"duration"`
+}
+
+type DialAttempt struct {
+	TargetAddr string `json:"targetAddr"`
+	Result     string `json:"result"`
+	Error      string `json:"error,omitempty"`
+	Duration   string `json:"duration"`
+}
+
+type EventsLogger struct {
+	fi    *os.File
+	enc   *json.Encoder
+	peers map[string]*PeerDialLog
+}
+
+func NewEventsLogger(path string) (*EventsLogger, error) {
+	f, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return nil, err
+	}
+
+	enc := json.NewEncoder(f)
+
+	return &EventsLogger{
+		fi:    f,
+		enc:   enc,
+		peers: make(map[string]*PeerDialLog),
+	}, nil
+}
+
+func (el *EventsLogger) handleEvents(r io.Reader) {
+
+	dec := json.NewDecoder(r)
+	for {
+		var ev map[string]interface{}
+		if err := dec.Decode(&ev); err != nil {
+			panic(err)
```

Comment:
we might want to not panic on EOF here -- maybe we want to be able to shutdown cleanly on some event in the future.

Reply:
Maybe log.Fatal is a better behaviour. I'll punt on checking for EOF, not sure what the semantics of this should be.

Reply:
I'll let it panic, we can reconsider if/when we hit it :)